ahcore/models/jit_model.py

```diff
@@ -1,24 +1,231 @@
+
+        Parameters
+        ----------
+        jit_path : Path
+            The path to the jit compiled model.
+
+        model_type: AllModelTypes
+            The type of the model.
+
+        output_mode: AllOutputModes
+            The output mode of the model. This is used to determine the forward function of the model.
+
+        Returns
+        -------
+        An instance of the AhcoreJitModel class.
+        """
         model = load(jit_path)  # type: ignore
-        return cls(model)
+        return cls(model, model_type, output_mode)
 
     def extend_model(self, modules: dict[str, Module]) -> None:
+        """
+        Add modules to a jit compiled model.
+
+        Parameters
+        ----------
+        modules : dict[str, Module]
+            A dictionary of modules to add to the model.
+
+        Returns
+        -------
+        None
+        """
         for key, value in modules.items():
             self._model.add_module(name=key, module=value)
 
     def forward(self, x: torch.Tensor) -> Any:
+        return self._forward_function(x)
+
+
+class FoundationModel(AhcoreJitModel):
+    """
+    This class is a wrapper for the foundation models in Ahcore.
+    It provides a general interface for the jit compiled foundation models that can be used in different contexts.
+    """
+    def __init__(self, model: ScriptModule, model_type: AllModelTypes, output_mode: AllOutputModes) -> None:
+        """
+        Constructor for the FoundationModel class.
+        A foundation model can be a feature extractor based on very large pre-trained model.
+
+        Parameters
+        ----------
+        model : ScriptModule
+            The jit compiled model.
+
+        model_type : AllModelTypes
+            The type of the model.
+
+        output_mode : AllOutputModes
+            The output mode of the model. This is used to determine the forward function of the model.
+
+        Returns
+        -------
+        None
+        """
+        super().__init__(model=model, model_type=JitModelType.FoundationModel, output_mode=output_mode)
+        self._fm_type = model_type
+        self._fm_embed_type = FMEmbedType.for_type(self._fm_type)
+        if self._output_mode == FMOutputMode.EMBED_CLS_ONLY:
+            self._forward_function = self._cls_only
+        elif self._output_mode == FMOutputMode.EMBED_PATCH_ONLY:
+            self._forward_function = self._patch_only
+        elif self._output_mode == FMOutputMode.EMBED_CONCAT:
+            self._forward_function = self._concat_embedding
+        elif self._output_mode == FMOutputMode.GENERAL_EMBEDDING:
+            self._forward_function = self._general_embedding
+        else:
+            raise NotImplementedError(f"Output mode {self._output_mode} is not supported for {self.__class__.__name__}")
+
+    @classmethod
+    def from_jit_path(cls, jit_path: Path, model_type: AllModelTypes, output_mode: AllOutputModes) -> Any:
+        model = load(jit_path)  # type: ignore
```

Comment:
Why does this need a type ignore?

Reply:
The torch.jit.load() is untyped. Mypy throws an error here if I don't ignore it.